protostar-rust/src/cheatcodes_hint_processor.rs

```diff
@@ -283,3 +401,71 @@ fn execute_cheatcode_hint(
         }
     }
 }
+
+fn create_execute_calldata(
+    calldata: &[Felt252],
+    class_hash: &ClassHash,
+    account_address: &ContractAddress,
+    entry_point_selector: &EntryPointSelector,
+    salt: &ContractAddressSalt,
+) -> Calldata {
+    let calldata_len = u128::try_from(calldata.len()).unwrap();
+    let mut execute_calldata = vec![
+        *account_address.0.key(),      // Contract address.
+        entry_point_selector.0,        // EP selector.
+        stark_felt!(calldata_len + 3), // Calldata length.
+        class_hash.0,                  // Calldata: class_hash.
+        salt.0,                        // Contract_address_salt.
+        stark_felt!(calldata_len),     // Constructor calldata length.
+    ];
+    let mut calldata: Vec<StarkFelt> = calldata
+        .iter()
+        .map(|data| StarkFelt::new(data.to_be_bytes()).unwrap())
+        .collect();
+    execute_calldata.append(&mut calldata);
+    Calldata(execute_calldata.into())
+}
+
+fn read_data_from_range(
```

Comment:
Can this one and below ones be implemented as a trait?

Reply:
Cairo already has a nice struct defined just for that https://github.com/starkware-libs/cairo/blob/main/crates/cairo-lang-runner/src/casm_run/mod.rs#L394 but I didn't want to copy all of the code to us.

Perhaps instead of making our own trait, we should make a PR to cairo repo that exposes this code as public?

Reply:
#2158